sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -759,6 +759,62 @@ static void az_span_copy_empty(void** state)
   assert_true(az_span_is_content_equal(az_span_copy(dst, AZ_SPAN_NULL), dst));
 }
 
+static void az_span_trim(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc")));
+}
+
+static void az_span_trim_left(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_start(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc   ")));
+}
+
+static void az_span_trim_right(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space_from_end(AZ_SPAN_FROM_STR("   abc   "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("   abc")));
+}
+
+static void az_span_trim_all_white(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("\t\n\r       "));
+  assert_int_equal(az_span_size(source), 0);
+}
+
+static void az_span_trim_none(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("abc"));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("abc")));
+}
+
+static void az_span_trim_spaced(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR("\ta\n b     c    "));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("a\n b     c")));
+}
+
+static void az_span_trim_zero(void** state)
+{
+  (void)state;
+  az_span source = _az_span_trim_white_space(AZ_SPAN_FROM_STR(""));
+  assert_true(az_span_is_content_equal(source, AZ_SPAN_FROM_STR("")));
```

Comment:
nit: Add calls to `_az_span_trim_white_space_from_end` and `_az_span_trim_white_space_from_start` here, in the `null` case, etc. to get more independent unit test coverage.

Reply:
what's the benefit of testing twice the same?

Reply:
You aren't testing the same twice. You are testing all of the public surface area, independently. The fact that trim calls trim_start/trim_end is an implementation detail. You shouldn't rely on that for code coverage.

We have 3 public APIs, ideally, the unit tests should be written to call each of the three.

Reply:
lol, I can add it, but, there is nothing different in this case from calling it from upper function.
Think I get what you say, but when functions are of the form of a -> b();  testing b() and testing a() is exactly the same as just testing a().
I would agree when the form of a is a -> {  c(); d(); b(); } because a(); is not a direct proxy of b();

Reply:
> but when functions are of the form of a -> b(); testing b() and testing a() is exactly the same as just testing a().

That relies too much on the **current** implementation of `a`. In the future, the implementation of `a` could change to no longer call `b()`, and in that case, you have lost test coverage.